src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,136 @@
+    private StorageDao storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUP() {
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_loginLessThanMinLength_notOk() {
+        User user = new User();
+        user.setLogin("Bob");
+        user.setPassword("ghjkl543");
+        user.setAge(23);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userAllInvalidFields_notOk() {
+        User user = new User();
+        user.setLogin("Micael");
+        user.setPassword("ghjkl");
+        user.setAge(15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_withTheSameFields_notOk() {
+        User user = new User();
+        user.setLogin("Micael");
+        user.setPassword("yxcvb123");
+        user.setAge(18);
+        registrationService.register(user);
+        User user1 = new User();
+        user1.setLogin("Micael");
+        user1.setPassword("yxcvb123");
+        user1.setAge(18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user1);
+        });
+    }
+
+    @Test
+    void register_withAgeLessThenMin_notOk() {
+        User user = new User();
+        user.setLogin("Micael");
+        user.setPassword("ghjkl1234");
+        user.setAge(12);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_withLogin_null_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("ghjkl1234");
+        user.setAge(17);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_passwordNull_notOk() {
+        User user = new User();
+        user.setLogin("Micael");
+        user.setPassword(null);
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_passwordLessThanMin_notOk() {
+        User user = new User();
+        user.setLogin("Micael");
+        user.setPassword("ghj");
+        user.setAge(18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userWithAllNormalFields_Ok() {
+        User user = new User();
+        user.setLogin("Micael");
+        user.setPassword("ghjkl1234");
+        user.setAge(25);
+        User normalUser = registrationService.register(user);
+        assertEquals(user, normalUser);
```

Comment:
After successful registration, you should verify that the user was actually added to the storage. Currently, the test only checks if the returned user is the same as the one registered, but not if it was stored.